source/common/upstream/conn_pool_map.h

```diff
@@ -46,6 +46,11 @@ template <typename KEY_TYPE, typename POOL_TYPE> class ConnPoolMap {
    */
   size_t size() const;
 
+  /**
+   * @return true if the pools are empty.
+   */
+  size_t empty() const;
```

Comment:
should this return `bool` instead of `size_t`?

Reply:
Yes, it definitely should. @lambdai can you post a followup to fix that?

Reply:
oops. Sure